### ISSUE TITLE
runtime: TestCgoPprofCallback hang on linux-arm

### ISSUE THREAD

**Issue by heschi** [golang/go#54778]
```
#!watchflakes
post <- pkg == "runtime" && test == "TestCgoPprofCallback"
```

[2022-08-26T18:09:56-62125c9/linux-arm-aws](https://build.golang.org/log/2dfc68933b0da9008a19e4c0c5c3b83c61f2a157)

The failure is giant so no summary here.

cc @golang/runtime

**Comment by mknyszek** [comment]
Looks like reproducing this with `GOTRACEBACK=crash` (or, do we already do that and it just didn't propagate to the test somehow?) will help; there are a bunch of `goroutine running on other thread; stack unavailable` goroutines that probably have some necessary information.

**Comment by mvdan** [comment]
Perhaps related: the test sometimes fails on linux-amd64-longtest, e.g. https://build.golang.org/log/a86dc70291e4f1c534e8798731dcec91160e5e99:

```
##### maymorestack=mayMoreStackPreempt
--- FAIL: TestCgoPprofCallback (0.67s)
    testenv.go:468: [/workdir/tmp/go-build1237047364/testprogcgo.exe CgoPprofCallback] exit status: signal: segmentation fault (core dumped)
    crash_cgo_test.go:228: expected "OK\n" got 
FAIL
FAIL	runtime	72.817s
```

**Comment by prattmic** [comment]
`greplogs --dashboard -md -l -e 'CgoPprofCallback'`

[2022-09-06T11:14:09-1c50484/linux-amd64-longtest](https://build.golang.org/log/f639653879210848179e32f2677c1fce64e20416)
[2022-09-05T21:39:28-4c1ca42/linux-amd64-longtest](https://build.golang.org/log/a86dc70291e4f1c534e8798731dcec91160e5e99)
[2022-09-05T08:28:34-bd5595d/linux-386-longtest](https://build.golang.org/log/3920217184f02a19abd0bb80ad17696dee0684ce)
[2022-09-05T08:12:37-67e6542/linux-amd64-longtest](https://build.golang.org/log/42665a73783912a2780243174431de7c1fb485d2)
[2022-09-05T08:07:47-357b922/linux-amd64-longtest](https://build.golang.org/log/ea8867f49a4106464cd73004b0e47c5ed6aab7c9)
[2022-09-04T04:17:04-535fe2b/linux-amd64-longtest](https://build.golang.org/log/d46ce2da64b4d2ae39905c55e2bc9dcdb2a9a059)
[2022-09-03T18:21:45-a0f0582/linux-386-longtest](https://build.golang.org/log/8256dfbc548facaaefaba5433c9bdf8c04b6fc44)
[2022-09-03T18:21:45-a0f0582/linux-amd64-longtest](https://build.golang.org/log/472e0ccf2c4eb10775d7ef026944e87e04be6826)
[2022-09-03T15:45:36-f798dc6/linux-amd64-longtest](https://build.golang.org/log/b875c95362f8b00744e1adcffb408d324a3333cc)
[2022-09-02T20:14:40-3e11e61/linux-amd64-longtest](https://build.golang.org/log/0096124f14e608f9df6ce2bd10a6af0417af2cd6)
[2022-09-02T20:14:16-646c3ee/linux-amd64-longtest](https://build.golang.org/log/cdcb5d009ce045dda84503451154a79e8171d5cf)
[2022-09-02T19:40:34-9154d4a/linux-amd64-longtest](https://build.golang.org/log/dfb9d6f3e30de1f405e972e28d25f7d2e4a363ab)
[2022-09-02T19:38:45-e406112/linux-amd64-longtest](https://build.golang.org/log/47db4df1679919a9449fb99f197d0fb9d9615831)
[2022-09-02T19:22:26-0fda8b1/linux-386-longtest](https://build.golang.org/log/c16de76490d3c7b5121c72089010926f7e3f5ccb)
[2022-09-02T19:08:56-b91e373/linux-amd64-longtest](https://build.golang.org/log/910e8ae6ac0075aba000000ad4cb5dddfe1f6d59)
[2022-09-02T19:08:53-dbf442b/linux-amd64-longtest](https://build.golang.org/log/20af38c170d1efb512c1f77289372007cce2ba82)

I suspect this sudden influx is actually a new issue related to #54885, and that the ARM issue is not quite the same.

**Comment by aclements** [comment]
This is not the same as https://github.com/golang/go/issues/54885, since that was introduced on 2022-09-02 and the original failure here happened well before that.

**Comment by mknyszek** [comment]
That one failure on linux/arm has been the only one since. Aside from a stream of persistent failures on longtest builders (that seem to be fixed), these are the latest 3:

[2022-11-09T22:56:44-89332e0/openbsd-amd64-68](https://build.golang.org/log/564884df382faa676a6b9bdad474e08935be1b51)
[2022-11-01T14:20:31-ad5d2f6/linux-amd64-longtest](https://build.golang.org/log/239d6bd0747bf1dbbb6b06924788c54d3a29ccfd)
[2022-11-01T14:11:54-2730170/openbsd-386-70](https://build.golang.org/log/38a07f1c6e5d62439bcff0a1b7999322461c1c1a)

**Comment by aclements** [comment]
This is marked "release-blocker" but is also in "Backlog." I'm not sure what the means, so I'm moving it to the 1.20 milestone.

**Comment by aclements** [comment]
> [2022-11-09T22:56:44-89332e0/openbsd-amd64-68](https://build.golang.org/log/564884df382faa676a6b9bdad474e08935be1b51)

This one appears to be unrelated.

**Comment by aclements** [comment]
One more failing TestCgoPprofCallback:
[2022-12-09T21:38:33-e8f78cb/openbsd-amd64-68](https://build.golang.org/log/6987aca69bc2ab13831782fa64aa8017d3c3a0f2)

This looks like a different bug, though.

**Comment by thanm** [comment]
This was discussed in the release weekly meeting-- it would be great to find out more about this bug so that we can re-evaluate it as a release blocker. Thanks.

**Comment by bcmills** [comment]
The `openbsd` failures in particular could be the general corruption bugs tracked in #55161 / #34988.

**Comment by heschi** [comment]
Ping -- we're getting close to the release.

**Comment by ianlancetaylor** [comment]
All the linux-amd64-longtest and linux-386-longtest failures are build with `-dmaymorestack=mayMoreStackPreempt`.  They also haven't happened for some time.

All the other failures appear to be on slow machines.  This is a CPU intensive test: it calls `runtime.GOMAXPROCS(16)` and then starts 64 goroutines.  Each goroutine does an endless loop of sleeping for 50 microseconds and calling a Go function.  The program keeps this up for a full second.  And, the test is run in parallel with other tests.

So I think this is a flaky test.  I think we should not return this CPU intensive test in parallel with other tests, and since it takes a full second I think we should not run it in short mode.

CC @prattmic for other opinions since he wrote the test.

**Comment by cherrymui** [comment]
The `mayMoreStackPreempt` ones are #54885, which is fixed.

**Comment by bcmills** [comment]
The `openbsd-.*-71` failures are probably #57585.
